### PR TITLE
Pd 3637 ui clear spinner and show error

### PR DESCRIPTION
**Pull request by aljordan82**
## What's this PR do?
Create a wrapper function for the Spinner. 
Any action that should show a Spinner should be wrapped with `wrapSpinnerAndErrors` or `wrapSpinner`.
## Any background context you want to provide?
This is just for Reporting, other react apps will need this change before merging this PR.
### Front-End

- [ ] Works on IE 11 and Edge
- [ ] Works on Mobile
- [ ] Unit tests added or ticket exists to add tests

### Separation of Concerns

- [ ] Back end "business logic" is implemented in plain python, Django model classes, and Django forms. (not Django views or other frameworks)
- [ ] Concerns of UI, api, and model are in separate code units.

### Tests

- [ ] Base classes used for python tests don't do too much unnecessary work.

### Migrations

- [ ] Migrations can be reversed.
- [ ] Current production code would continue to work after this migration is applied.
- [ ] Data migrations' forward/reverse functions should be decorated with `@only_on_default`.

### CODE REVIEW

File: gulp/src/reporting/actions/compound-actions.js [hunk 1]
```diff
@@ -151,10 +150,9 @@ export function doReportDataRedirect(history, intention, category, dataSet) {
       dispatch(markOtherLoadingAction('dataSetMenu', false));
     } catch (e) {
       dispatch(markOtherLoadingAction('dataSetMenu', false));
-      dispatch(markPageLoadingAction(false));
-      dispatch(errorAction(e.message));
+      throw e;
```
Comment (aljordan82) [inline gulp/src/reporting/actions/compound-actions.js]: If the wrapped action creator has a try/catch , then we must throw so the wrapper handles the error.

**Comment by koonsies** [comment]
I like this idea. While we're refactoring the handling of spinner, etc, perhaps we could revisit how we "mark" a page as loading. One method I've seen was in CBO, and that was to store a counter of functions being waited on. So, if you dispatched three functions that were wrapped in the spinner wrapper, each would increment a global function counter by 1. When a function completed (or errored out), it would decrement the counter until the counter reached 0 and the spinner was removed.

We may have discussed this before, but I can't recall. Just wanted to bounce that idea around again.

**Comment by vigowei** [comment]
maybe this is also an opportunity to agree on the format for error messages? Sometimes the error string is at the top level of the response object, so we can get it via response.data, sometimes, it's in an object, like `{'message': 'reason for failure...'}`, now we have to pass response.data.message to toastr since it doesn't display objects. 

we could stringify/format whatever is in response.data, or designate a field for error message, which means making sure all error responses from the backend follow the same convention.

**Comment by koonsies** [comment]
Closing this until it's ready to review again.

**Pull request closed without merging**